### PR TITLE
[Bugfix] Allowing Editors to trigger Instructor Builds

### PR DESCRIPTION
**Pull request by niclasschuemann**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.

### Motivation and Context
At the moment, editors cannot trigger the instructor build for template and solution participation in programming exercises, even though they should be able to do it and have access to the respective button on the exercise detail page.

### Description
I have added a new check that makes sure that editors can issue instructor builds for template and solution participation, but not for student participation.

### Steps for Testing

**You will need:** 1x TA, 1x Editor, 1x instructor, 1x Programming Exercise, 1x Student Submission on this Exercise

1. With the TA account: Go to the programming exercise detail view. Make sure that the resubmit buttons for template and solution participation are not present.
2. Repeat step 1. with the Editor account and make sure that the buttons are now present.
3. With the Editor, click on both buttons and make sure that the participation status changes to "Building and testing"
4. Wait for the result and make sure that a result is actually being generated.
5. Open the template and solution build plan on Bamboo and make sure that the build was actually triggered
6. With the Editor, go to the Participation View of the programming exercise. Make sure that no resubmit button is present.
7. Repeat step 6. with the Instructor. Make sure that the button is present and actually works.

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingSubmissionResource.java [hunk 1]
```diff
@@ -146,8 +148,16 @@ public ProgrammingSubmissionResource(ProgrammingSubmissionService programmingSub
...
             return notFound();
         }
-        if (!programmingExerciseParticipationService.canAccessParticipation(programmingExerciseParticipation)
-                || (submissionType.equals(SubmissionType.INSTRUCTOR) && !authCheckService.isAtLeastInstructorForExercise(participation.getExercise()))) {
+
+        if (!programmingExerciseParticipationService.canAccessParticipation(programmingExerciseParticipation)) {
+            return forbidden();
+        }
+
+        // The editor is allowed to trigger an instructor type build for template and solution participations,
```
Comment (FlorianGlombik) [inline src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingSubmissionResource.java:156]:
```suggestion
        // The editor is allowed to trigger an instructor build for template and solution participations,
```
A matter of taste, can be ignored.

**Review by joschahenningsen — DISMISSED** [review]
worked for me :+1:

**Review by FlorianGlombik — APPROVED** [review]
Tested it on ts1, the buttons work for instructors and editors, builds were triggered.

Tutors can not see the buttons, which are intended.

**Review by PatZintel — APPROVED** [review]
worked well on ts3, both editor and instructor can see and use the trigger builds buttons and they actually work (even though for me only after reloading the page, the result of the build gets shown, otherwise i am stuck on Building an Testing). The Tutor couldnt see the buttons.

**Review by JohannesStoehr — APPROVED** [review]
The code changes look good as well. Nice work 👍

**Review by ivanchimeno — APPROVED** [review]
Works on Ts1 :+1:

**Pull request merged**